src/Firebase.js

```diff
@@ -8,55 +6,54 @@ import { createQueryRef, mapValues, mapSnapshotToValue, pickBy } from './utils'
 const mapSubscriptionsToQueries = subscriptions =>
   mapValues(subscriptions, value => (typeof value === 'string' ? { path: value } : value))
 
-const getQuery = (props, ref) => {
+const getQuery = (props, ref, app) => {
   switch (typeof props.query) {
     case 'string':
       return { [props.query]: props.query }
     case 'function':
-      return props.query(ref)
+      return props.query(ref, app)
     default:
       return props.query
   }
 }
 
-const getSubscriptions = (props, ref) =>
-  pickBy(getQuery(props, ref), prop => typeof prop === 'string' || (prop && prop.path))
+const getSubscriptions = (props, ref, app) =>
+  pickBy(getQuery(props, ref, app), prop => typeof prop === 'string' || (prop && prop.path))
 
-const getActions = (props, ref) => pickBy(getQuery(props, ref), prop => typeof prop === 'function')
+const getActions = (props, ref, app) =>
+  pickBy(getQuery(props, ref, app), prop => typeof prop === 'function')
 
 export default class Firebase extends React.Component {
   static propTypes = {
     render: PropTypes.func.isRequired,
     query: PropTypes.oneOfType([PropTypes.object, PropTypes.string, PropTypes.func]).isRequired,
-    firebaseApp: PropTypes.shape({
-      database: PropTypes.func.isRequired,
-    }),
+    firebaseApp: PropTypes.instanceOf(firebase.app.App), // eslint-disable-line react/no-unused-prop-types
   }
 
   static contextTypes = {
-    firebaseApp: PropTypes.shape({
-      database: PropTypes.func.isRequired,
-    }),
+    firebaseApp: PropTypes.instanceOf(firebase.app.App),
   }
 
   state = {
-    subscriptionsState: null,
+    subscriptionsState: {},
```

Comment:
What's the reason for using an empty object over `null` here? Is this because there's no null-check at 87?

Reply:
Yes